consensus/tendermint/wal_test.go

```diff
@@ -0,0 +1,186 @@
+package tendermint
+
+import (
+	"testing"
+
+	db "github.com/NethermindEth/juno/consensus/db"
+	"github.com/NethermindEth/juno/consensus/mocks"
+	"github.com/NethermindEth/juno/consensus/types"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/utils"
+	"go.uber.org/mock/gomock"
+)
+
+func getPrevote(idx int) types.Prevote[felt.Felt, felt.Felt] {
+	return types.Prevote[felt.Felt, felt.Felt]{
+		MessageHeader: types.MessageHeader[felt.Felt]{
+			Height: types.Height(0),
+			Round:  types.Round(0),
+			Sender: *getVal(idx),
+		},
+		ID: utils.HeapPtr(felt.FromUint64(1)),
+	}
+}
+
+func getPrecommit(idx int) types.Precommit[felt.Felt, felt.Felt] {
+	return types.Precommit[felt.Felt, felt.Felt]{
+		MessageHeader: types.MessageHeader[felt.Felt]{
+			Height: types.Height(0),
+			Round:  types.Round(0),
+			Sender: *getVal(idx),
+		},
+		ID: utils.HeapPtr(felt.FromUint64(1)),
+	}
+}
+
+func TestReplayWAL(t *testing.T) {
+	proposerAddr := getVal(0)
+	nonProposerAddr := getVal(3)
+	proposedValue := value(1)
+	app, chain, vals := newApp(), newChain(), newVals()
+	for i := range 4 {
+		vals.addValidator(*getVal(i))
+	}
+
+	ctrl := gomock.NewController(t)
+	proposalMessage := types.Proposal[value, felt.Felt, felt.Felt]{
+		MessageHeader: types.MessageHeader[felt.Felt]{
+			Height: types.Height(0),
+			Round:  types.Round(0),
+			Sender: *proposerAddr,
+		},
+		ValidRound: types.Round(-1),
+		Value:      &proposedValue,
+	}
+
+	t.Run("ReplayWAL: replay on empty db", func(t *testing.T) {
+		mockDB := mocks.NewMockTendermintDB[value, felt.Felt, felt.Felt](ctrl)
```

Comment:
I think in this specific case, we should test against a real DB. The test cases can be in a simple format:
Do something -> Crash -> Restart -> Assert if the new state and the actions are the same as the old ones.